rpcs3/xinput_pad_handler.cpp

```diff
@@ -2,12 +2,222 @@
+
+private:
+	PFN_XINPUTGETSTATE xinputGetState{nullptr};
+
+public:
+	virtual pad_handler GetHandlerType() const override
+	{
+		return pad_handler::xinput;
+	}
+
+	virtual const char* GetNameString() const override
+	{
+		return "XInput Pad #";
+	}
+
+	virtual const std::unordered_map<u32, std::string>& GetButtonList() const override
+	{
+		static const std::unordered_map<u32, std::string> button_list =
+		{
+			{ xinput_pad_handler::XInputKeyCodes::A,      "A" },
+			{ xinput_pad_handler::XInputKeyCodes::B,      "B" },
+			{ xinput_pad_handler::XInputKeyCodes::X,      "X" },
+			{ xinput_pad_handler::XInputKeyCodes::Y,      "Y" },
+			{ xinput_pad_handler::XInputKeyCodes::Left,   "Left" },
+			{ xinput_pad_handler::XInputKeyCodes::Right,  "Right" },
+			{ xinput_pad_handler::XInputKeyCodes::Up,     "Up" },
+			{ xinput_pad_handler::XInputKeyCodes::Down,   "Down" },
+			{ xinput_pad_handler::XInputKeyCodes::LB,     "LB" },
+			{ xinput_pad_handler::XInputKeyCodes::RB,     "RB" },
+			{ xinput_pad_handler::XInputKeyCodes::Back,   "Back" },
+			{ xinput_pad_handler::XInputKeyCodes::Start,  "Start" },
+			{ xinput_pad_handler::XInputKeyCodes::LS,     "LS" },
+			{ xinput_pad_handler::XInputKeyCodes::RS,     "RS" },
+			{ xinput_pad_handler::XInputKeyCodes::Guide,  "Guide" },
+			{ xinput_pad_handler::XInputKeyCodes::LT,     "LT" },
+			{ xinput_pad_handler::XInputKeyCodes::RT,     "RT" },
+			{ xinput_pad_handler::XInputKeyCodes::LSXNeg, "LS X-" },
+			{ xinput_pad_handler::XInputKeyCodes::LSXPos, "LS X+" },
+			{ xinput_pad_handler::XInputKeyCodes::LSYPos, "LS Y+" },
+			{ xinput_pad_handler::XInputKeyCodes::LSYNeg, "LS Y-" },
+			{ xinput_pad_handler::XInputKeyCodes::RSXNeg, "RS X-" },
+			{ xinput_pad_handler::XInputKeyCodes::RSXPos, "RS X+" },
+			{ xinput_pad_handler::XInputKeyCodes::RSYPos, "RS Y+" },
+			{ xinput_pad_handler::XInputKeyCodes::RSYNeg, "RS Y-" }
+		};
+		return button_list;
+	}
+
+	bool Init() override
+	{
+		if (is_init)
+			return true;
+
+		for (auto it : LIBRARY_FILENAMES)
+		{
+			library = LoadLibrary(it);
+			if (library)
+			{
+				xinputGetState = reinterpret_cast<PFN_XINPUTGETSTATE>(GetProcAddress(library, reinterpret_cast<LPCSTR>(100)));
+				if (!xinputGetState)
+					xinputGetState = reinterpret_cast<PFN_XINPUTGETSTATE>(GetProcAddress(library, "XInputGetState"));
+
+				xinputSetState              = reinterpret_cast<PFN_XINPUTSETSTATE>(GetProcAddress(library, "XInputSetState"));
+				xinputGetBatteryInformation = reinterpret_cast<PFN_XINPUTGETBATTERYINFORMATION>(GetProcAddress(library, "XInputGetBatteryInformation"));
+
+				if (xinputGetState && xinputSetState && xinputGetBatteryInformation)
+				{
+					is_init = true;
+					break;
+				}
+
+				FreeLibrary(library);
+				library                     = nullptr;
+				xinputGetState              = nullptr;
+				xinputGetBatteryInformation = nullptr;
+			}
+		}
+
+		if (!is_init)
+			return false;
+
+		return true;
```

Comment:
return is_init

Reply:
Was like this in old code too, but I'll change.

Reply:
it was probably just added for a breakpoint